src/bvh/optimization.rs

```diff
@@ -288,9 +298,7 @@ impl<T: BHValue, const D: usize> Bvh<T, D> {
             if deleted_shape_index < end_shape {
                 shapes.swap(deleted_shape_index, end_shape);
                 let node_index = shapes[deleted_shape_index].bh_node_index();
-                if let Some(index) = self.nodes[node_index].shape_index_mut() {
-                    *index = deleted_shape_index
-                }
+                *self.nodes[node_index].shape_index_mut().unwrap() = deleted_shape_index;
```

Comment:
Is it safe to unwrap this? Are we absolutely sure this can never crash? If so, please comment why this is the case. If there's a remote possibility that it might crash, use `expect()`.

Reply:
`Option::unwrap` is analogous to `slice[index]`. They'll both panic if expectations are violated (the BVH is inconsistent, which should never happen). To be clear, are you suggesting to only comment the occasional `unwrap` but not the very common `[..]`?

I can certainly comment this, but it feels inconsistent.

Reply:
That's a good point actually. Perhaps we are in fact too liberal with `[]` where we should be using `.get()` with a proper error? I don't currently have a good suggestion here since we use `[]` a whole lot. At any rate, this works for now. :)